lib/puppet/resource.rb

```diff
@@ -168,10 +168,15 @@ def is_json_type?(value)
   end
 
   def self.value_to_pson_data(value)
-    if value.is_a? Array
+    case value
+    when Array
       value.map{|v| value_to_pson_data(v) }
-    elsif value.is_a? Puppet::Resource
+    when Hash
+      Hash[ value.map {|k, v| [value_to_pson_data(k), value_to_pson_data(v)] } ]
```

Comment:
This change renders the `case` statement in the method `yaml_property_munge(x)` below, obsolete.